compose.base.yaml

```diff
@@ -20,3 +20,5 @@ services:
       REDASH_COOKIE_SECRET: ${REDASH_COOKIE_SECRET}
       REDASH_SECRET_KEY: ${REDASH_SECRET_KEY}
       REDASH_PRODUCTION: ${REDASH_PRODUCTION:-true}
+    env_file:
+      - .env
```

Comment:
```suggestion
    env_file:
      - path: .env
         required: false
```

Reply:
Is there another way around this? It appears that making `env_file` optional was not available until docker-compose 2.24.0

https://docs.docker.com/compose/environment-variables/set-environment-variables/#additional-information

So making this change would introduce a new dependency.

Reply:
> docker compose 2.24 is available since january, think people had enough time to upgrade.

It's not always that simple.

We've previously had to freeze the version of docker, due to a bug (in docker) that stopped our Dockerfile from working. :frowning:

Reply:
Looking at our options though, `required: false` is probably worth trying for now and seeing if causes problems for anyone?

Reply:
Well, it caused problems for me since I usually use docker-compose-v2 proved by Ubuntu 22.04 packages.

The latest version of docker-compose-v2 can, of course be installed separately, but this adds complexity to the existing setup instructions.

Reply:
No worries, looks like the `required: false` thing isn't really the way forward then.

We'd better add something (script?) which gets the .env working for the CI environment too. :smile:

Reply:
I updated this PR to run ` touch .env` before building the server in `ci.yml`. I tested this in my repo, and it seemed to work.

Reply:
The concept (of `touch .env`) seems good, but I'm guessing it's somehow doing that in the wrong directory or something from the CI point of view:

```
+ docker compose build
Failed to load /home/runner/work/redash/redash/.env: open /home/runner/work/redash/redash/.env: no such file or directory
```

Might be worth adding a bunch of extra locations to the touch (`touch redash/.env redash/redash/.env /home/runner/work/redash/redash/.env`, etc) to see if it gets the CI to pass?  And if that works, then figure out which one of the paths fixed it.

Or maybe embed a `pwd` in there or something, as that might clear things up too? :smile:

Reply:
almost everywhere where it's possible to find docker setup instructions, it's recommended to use official repo even sometimes it's recommended to remove old docker packages before installing official ones. there's no problem to mention somewhere minimal required docker version. there should be no problems with upgrade for each user on each os

Reply:
@AndrewChubatiuk There are different schools of thought of that.  Our `setup.sh` uses the Docker CE packages, but our "Local development setup" instructions use the Ubuntu provided packages instead.

We should probably choose one and use that for both.

I'm personally more inclined to use the Docker CE ones, as we've previously seen bugs in the Ubuntu provided ones.

... though I've also seen bugs on the Docker CE ones historically as well. :facepalm:

Reply:
> The concept (of `touch .env`) seems good, but I'm guessing it's somehow doing that in the wrong directory or something from the CI point of view:

No, the problem is that the change to the CI build will not take effect until this change is merged.

Reply:
Ahhh.  So we should force merge this then?  We can do that if needed.

Reply:
Yes, we should skip the checks and merged this. Hard to be 100% sure, but I did test this in my own account

Reply:
Cool, all good. :smile: